src/server/lib/idapi/consents.ts

```diff
@@ -67,3 +68,49 @@ export const update = async (
     return handleError();
   }
 };
+
+export const getUserConsentsForPage = async (
+  pageConsents: string[],
+  ip: string,
+  sc_gu_u: string,
+): Promise<Consent[]> => {
+  const allConsents = await read();
+  const userConsents = (await getUser(ip, sc_gu_u)).consents;
+
+  return pageConsents
+    .map((id) => allConsents.find((consent) => consent.id === id))
+    .map((consent) => {
+      if (consent) {
+        let updated = consent;
+        const userConsent = userConsents.find((uc) => uc.id === consent.id);
+
+        if (userConsent) {
+          updated = {
+            ...updated,
+            consented: userConsent.consented,
+          };
+        }
+
+        return updated;
+      }
+    })
+    .filter(Boolean) as Consent[];
+};
+
+export const getConsentValueFromRequestBody = (
+  key: string,
+  body: { [key: string]: string },
+): boolean => {
+  if (body[key] === undefined || typeof body[key] !== 'string') {
+    return false;
+  }
+
+  switch (body[key]) {
```

Comment:
I wonder if this switch block could just be replaced with `return !!body[key];`

Reply:
I don't think so, as `!!'false'` would return `true`.

This and the above function have been moved from `src/server/routes/consents.ts` btw, I haven't actually changed them at all. Just moved them here as they are needed in the sign in routes now also.

Reply:
ah yes, didn't notice it was a string. all good!